Tempura/Core/ViewController+Containment.swift

```diff
@@ -16,10 +16,43 @@ extension ViewController {
     view.addSubview(child.rootView)
     child.didMove(toParent: self)
   }
-  
+
+  /// Replace a ViewController (and its rootView) to the last child of self.
+  /// You must provide a ContainerView inside self.rootView, the latest view of that container view
+  /// is replaced with the rootView of the child ViewController with a cross-dissolve transition
+  public func transition<View: ViewControllerModellableView>(
+    to child: ViewController<View>,
+    in view: ContainerView,
+    duration: Double = 0.3,
+    options: UIView.AnimationOptions = [],
+    completion: (() -> ())? = nil
+  ) {
+    guard
+      let lastView = view.subviews.last,
+      let lastViewVC = (lastView as? AnyViewControllerModellableView)?.viewController
+    else {
+      return
+    }
+
+    UIView.transition(
+      from: lastView,
+      to: child.rootView,
+      duration: duration,
+      options: [.transitionCrossDissolve],
+      completion: { _ in
+        self.addChild(child)
+        child.didMove(toParent: self)
```

Comment:
Don't we need to actually add child as subview?

Reply:
It should be done by the UIView.transition: https://developer.apple.com/documentation/uikit/uiview/1622562-transition

```
fromView
The starting view for the transition. By default, this view is removed from its superview as part of the transition.

toView
The ending view for the transition. By default, this view is added to the superview of fromView as part of the transition.
```

Reply:
Cool! 🙌